packages/adminrouter/extra/src/common/auth/common.lua

```diff
@@ -32,53 +28,36 @@ else
     end
 end
 
-
-local key_file_path = os.getenv("SECRET_KEY_FILE_PATH")
-if key_file_path == nil then
-    ngx.log(ngx.WARN, "SECRET_KEY_FILE_PATH not set.")
-else
-    ngx.log(ngx.NOTICE, "Reading secret key from `" .. key_file_path .. "`.")
-    SECRET_KEY = util.get_stripped_first_line_from_file(key_file_path)
-    if (SECRET_KEY == nil or SECRET_KEY == '') then
-        -- Normalize to nil, for simplified subsequent per-request check.
-        SECRET_KEY = nil
-        ngx.log(ngx.WARN, "Secret key not set or empty string.")
-    end
-	jwt:set_alg_whitelist({HS256=1})
-end
-
-
--- Refs:
--- https://github.com/openresty/lua-nginx-module#access_by_lua
--- https://github.com/SkyLothar/lua-resty-jwt
-
-
-local function exit_401()
+local function exit_401(authtype)
     ngx.status = ngx.HTTP_UNAUTHORIZED
     ngx.header["Content-Type"] = "text/html; charset=UTF-8"
-    ngx.header["WWW-Authenticate"] = "oauthjwt"
+    ngx.header["WWW-Authenticate"] = authtype
     ngx.say(BODY_401_ERROR_RESPONSE)
     return ngx.exit(ngx.HTTP_UNAUTHORIZED)
 end
 
-
 local function exit_403()
     ngx.status = ngx.HTTP_FORBIDDEN
     ngx.header["Content-Type"] = "text/html; charset=UTF-8"
     ngx.say(BODY_403_ERROR_RESPONSE)
     return ngx.exit(ngx.HTTP_FORBIDDEN)
 end
 
-
-local function validate_jwt_or_exit()
+-- We need to differentiate how secret_key value is acquired so we acquire it
+-- in different lua fila. This requires either passig this value as a global or
+-- as a function argument. Function argument seems like a safer approach.
+local function validate_jwt(secret_key)
     -- Inspect Authorization header in current request. Expect JSON Web Token in
     -- compliance with RFC 7519. Expect `uid` key in payload section. Extract
-    -- and return uid. In all other cases, terminate request handling and
-    -- respond with an appropriate HTTP error status code.
+    -- and return uid or the error code.
 
-    if SECRET_KEY == nil then
+    -- Refs:
+    -- https://github.com/openresty/lua-nginx-module#access_by_lua
+    -- https://github.com/SkyLothar/lua-resty-jwt
+
+    if secret_key == nil then
         ngx.log(ngx.ERR, "Secret key not set. Cannot validate request.")
-        return exit_401()
+        return nil, 401
```

Comment:
Having two distinct vars returned by the function is I think safer - if for some weird reason we do not catch error condition, and uid ends up nil - it will be immediately visible/easy to spot. Having second value being True/False instead of 401/HTTP code is not as readable I think. Thus I explicitly return "401" so that person reading the code can say - `Ah, so the result of the validation was "unauthenticated"!`. 

If/when we add more error conditions to the `validate_jwt`, we also get the default catch-all which verifies that the error is, in fact, one of the expected ones and thus we get even better error checking.